akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/aggregate/GeneralAggregateBase.scala

```diff
@@ -47,6 +48,36 @@ abstract class GeneralAggregateBase[E:ClassTag, S <: AggregateStateBase[E, S]:Cl
   private val defaultSuccessHandler = () => log.debug("No cmdSuccess-handler executed")
   private val defaultErrorHandler = (errorMsg:String) => log.debug("No cmdFailed-handler executed")
 
+  //Overridable handling of snapshot related messages
+  protected val aggregatePersistenceHandling : AggregateSnapshotHandler = new AggregateSnapshotHandler {

```

Comment:
I don't like the use of this AggregateSnapshotHandler.

I think all needed method should be implemented directly in class.. And that user should be able to override what he wants.

But I'm not sure how much / what hould be possible to override..

Reply:
Sure. The idea was to ensure the user evaluated all of this when implementing.